modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -22,7 +23,52 @@ package engine {
     def fromProcess(p: Process[Task, Event]): HandleP[Unit] = HandleP(Applicative[Handle].pure[(Unit, Option[Process[Task, Event]])](((), Some(p))))
   }
   final case class ActionMetadata(conditions: Conditions, operator: Option[Operator], observer: Option[Observer])
-  final case class Action(kind: ActionType, gen: Kleisli[Task, ActionMetadata, Result])
+  object ActionMetadata {
+    val default: ActionMetadata = ActionMetadata(Conditions.default, None, None)
+  }
+
+  final case class Action(
+    kind: ActionType,
+    gen: ActionGen,
```

Comment:
`ActionGen`?

Reply:
It is an alias for `Kleisli[Task, ActionMetadata, Result]`, the function that generates the `Task` that is the body of the `Action`.